File: server/src/main/java/com/businessbear/server/entity/RealAsset.java
package com.businessbear.server.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "real_assets")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class RealAsset {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, length = 50)
    private String code;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, length = 255)
    private String location;

    @Column(nullable = false, precision = 15, scale = 2)
    private BigDecimal price;

    @Column(name = "image_url", length = 512)
    private String imageUrl;

    private Integer beds;

    private Double baths;

    private Integer sqft;

    @Enumerated(EnumType.STRING)
    @Column(length = 30, nullable = false)
    @Builder.Default
    private AssetStatus status = AssetStatus.FOR_SALE;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "is_featured", nullable = false)
    @Builder.Default
    private Boolean isFeatured = false;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    public void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: gui/types/real-asset.ts
export type AssetStatus =
  | "FOR_SALE"
  | "FOR_RENT"
  | "NEW_LISTING"
  | "JUST_SOLD"
  | "BOOKED";

export const ASSET_STATUS_LABELS: Record<AssetStatus, string> = {
  FOR_SALE: "For Sale",
  FOR_RENT: "For Rent",
  NEW_LISTING: "New Listing",
  JUST_SOLD: "Just Sold",
  BOOKED: "Booked",
};

export interface RealAsset {
  id?: number;
  code?: string;
  title: string;
  location: string;
  price: number;
  imageUrl?: string;
  beds?: number | null;
  baths?: number | null;
  sqft?: number | null;
  status?: AssetStatus;
  description?: string;
  isFeatured?: boolean;
}
